### ISSUE TITLE
Parsing of expression failed

### ISSUE THREAD

**Issue by pitrou** [michaeljones/breathe#788]
When building the Apache Arrow docs, we get this warning:
```
/home/antoine/arrow/dev/docs/source/cpp/api/dataset.rst:62: WARNING: Parsing of expression failed. Using fallback parser. Error was:
Error in postfix expression, expected primary expression or type.
If primary expression:
  Invalid C++ declaration: Expected identifier in nested name. [error at 59]
    std::function< Status(FileWriter *)> writer_pre_finish   = [](FileWriter*) {returnStatus::OK();}
    -----------------------------------------------------------^
If type:
  Invalid C++ declaration: Expected identifier in nested name. [error at 59]
    std::function< Status(FileWriter *)> writer_pre_finish   = [](FileWriter*) {returnStatus::OK();}
    -----------------------------------------------------------^

```

Admittedly, C++ is hard to parse, so not necessarily expecting this to pass :-) Still, the warning is a bit distracting.

**Comment by jakobandersen** [comment]
Right, this is not a problem Breathe can solve, but is for Sphinx.
As you may have guessed, the problem is that parsing of lambda expressions has not been implemented yet, as that basically means _all_ of C++ can be encountered.
Note that the rendering of the declaration should still be fine, except for the lambda expression being without styling.
Please open an issue over at Sphinx, then I'll progress on in when time allows. If you have other examples of such lambda expressions that are not being parsed, please list them as well. Then I'll prioritise the statement types that you need.

**Comment by pitrou** [comment]
To be honest, we probably don't care whether the lambda expression is properly syntax-highlighted or not. What's mostly annoying is the warning itself :-)

**Comment by pitrou** [comment]
Sphinx issue open at https://github.com/sphinx-doc/sphinx/issues/10152
